recipes/cyclonedds/all/conanfile.py

```diff
@@ -0,0 +1,174 @@
+            self.requires("iceoryx/2.0.0")
+        if self.options.ssl:
+            self.requires("openssl/1.1.1q")
+    
+    def build_requirements(self):
+        self.tool_requires("cmake/3.16.2")
+
+    def validate(self):
+        compiler = self.settings.compiler
+        version = scm.Version(self.settings.compiler.version)
+
+        if ((self.options.security is True ) and (self.options.shared is False)):
+            raise ConanInvalidConfiguration("Cyclone DDS currently do not support"\
+                                            "static build and security on")
+
+        if compiler.get_safe("cppstd"):
+            build.check_min_cppstd(self, 14)
+
+        if compiler == "Visual Studio":
+            # ToDo : determine windows error and find solution (at test_package)
+            raise ConanInvalidConfiguration("Cyclone DDS is not (yet) supported"\
+                                                "for Visual Studio compiler.")
+        elif compiler == "gcc":
+            if version < "6":
+                raise ConanInvalidConfiguration("Using Cyclone DDS with gcc requires"\
+                                                " gcc 6 or higher.")
+            if version < "9" and compiler.get_safe("libcxx") == "libstdc++":
+                raise ConanInvalidConfiguration("gcc < 9 with libstdc++ not supported")
+            if version == "6":
+                self.output.warn("Cyclone DDS package is compiled with gcc 6, it is"\
+                                 " recommended to use 7 or higher")
+                self.output.warn("GCC 6 will build with warnings.")
+        elif compiler == "clang":
+            if compiler.get_safe("libcxx") == "libstdc++":
+                raise ConanInvalidConfiguration("clang with libstdc++ not supported")
+            if version == "7.0" and compiler.get_safe("libcxx") == "libc++" and \
+               self.options.shared and self.settings.build_type == "Debug":
+                raise ConanInvalidConfiguration("shared Debug with clang 7.0 and"\
+                                                " libc++ not supported")
+
+    def source(self):
+        files.get(self,**self.conan_data["sources"][self.version], strip_root=True,
+                 destination=self._source_subfolder)
+
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+
+        tc = CMakeToolchain(self)
+        # ToDo : determine how to do in conan : 
+        # - idlc is a code generator that is used as tool (and so not cross compiled)
+        # - other tools like ddsperf is cross compiled for target
+        # - maybe separate package like cyclonedds_idlc 
+        tc.variables["BUILD_IDLC"]            = False
+        tc.variables["BUILD_IDLC_TESTING"]    = False
+        tc.variables["BUILD_DDSPERF"]         = False
+        tc.variables["BUILD_IDLC_TESTING"]    = False
+        # variables which effects build
+        tc.variables["BUILD_SHARED_LIBS"]     = self.options.shared
```

Comment:
```suggestion
```
Automatically managed: https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmaketoolchain.html#extending-and-customizing-cmaketoolchain